api/watch/funcs_test.go

```diff
@@ -300,6 +300,8 @@ func TestServicesWatch(t *testing.T) {
 	c, s := makeClient(t)
 	defer s.Stop()
 
+	s.WaitForSerfCheck(t)
```

Comment:
Upon looking at the cases in this file, I think if you were going to wait for the agent serf check for this one, any node or agent-oriented watch should also wait for similar AE reasons:
- `TestServicesWatch
- `TestServiceWatch
- `TestServiceMultipleTagsWatch
- `TestChecksWatch_State`
- `TestChecksWatch_Service`
- `TestConnectLeafWatch`
- `TestAgentServiceWatch`

For reference I just found any test method in this file that made reference to a `api.AgentYYYYYY` struct and just assumed it would be sane to wait for that agent to fully be represented in the server's state before continuing.

It should just in general be safe to wait for the serf check in all of these watch tests since they all startup about right.

Reply:
Added these, searching for more potential places with `rg api.Agent -g "*_test.go"` turned up a bunch of matches, but it feels like maybe we should just merge this before trying to tackle a bunch more?